research-software-directory/RSD-as-a-service
# 1294 categories for organisations

# Categories for organisations

Closes #1294

Changes proposed in this pull request:
* Organisation maintainer can create custom categories to be used by projects and software of the organisation
* Software and project maintainers can select custom categories specific to organisation      
* Organisation categories are shown on software page in the side panel on the left, under the logo
* Organisation categories are shown on project page in the side panel on the left
* Refactored code to reduce code duplication. ~5% is optimal amount in my opinion, otherwise we will need additional level of abstraction which in my opinion will further increase cognitive load.
* Updated user documentation (software,project and organisation pages)

How to test:
* `make start` to build and generate test data
* navigate to organisations and select first organisation
* login as rsd-admin (first login in dev mode)
* navigate to organisation settings, categories tab
* add categories to organisation:
   * create project specific categories (use "For projects" switch)
   * create software specific categories
   * create shared categories (check both for software and project switch)
* navigate to software item and add organisation that you just added the categories to
* confirm that categories modal appears and you can assign organisation categories 
* select some categories and confirm that these are shown on the software page
* navigate to project item and add organisation that you just added the categories to
* confirm that categories modal appears and you can assign organisation categories 
* select some categories and confirm that these are shown on the project page
* **bonus test:** enable communities in the settings.json and test community categories for software

## Example organisation categories
![image](https://github.com/user-attachments/assets/a781112a-cb48-4614-922b-db478c42f3f6)

## Example adding organisation categories to software or project
![image](https://github.com/user-attachments/assets/0a691479-47ad-4e98-b7d9-0dd647a48f0a)

## Example organisation categories on project page (bottom of side panel)
![image](https://github.com/user-attachments/assets/ac524bbf-c923-4610-bb52-1a0c0f545a28)

PR Checklist:
